packages/controlled-form/src/lib/Types.tsx

```diff
@@ -33,6 +33,17 @@ export type ControllerProps = {
   shouldUnregister?: MuiControllerProps['shouldUnregister'];
 };
 
+export type TransformProp<Input = string, Output = string> = {
+  /** Object containing functions to parse the input value and format the output value. Useful for converting values from API responses to display values and back.
+   * - `input`: Function to transform the value from the underlying data to the input value.
+   * - `output`: Function to transform the value from the input to the underlying data.
+   * 
+   * Example: The input always expects a dayjs object, but your endpoint expects and returns a string:
+   * `{ input: (value) => value ? dayjs(value) : null, output: (value) => value.format('YYYY-MM-DD') }`
+   */
+  transform?: { input?: (value: Output) => Input; output?: (value: Input) => Output };
+};
+
 // Storybook Categories
 
 type AllControllerProps = ControllerProps & Pick<RegisterOptions<FieldValues, string>, 'onBlur' | 'onChange' | 'value'>;
```

Comment:
All rhf props should be included in AllControllerProps for proper categorization in storybook.

Reply:
Wasn't sure if these were really considered controller props since it doesn't go on the controller, but instead goes on each of the rendered inputs.

I updated the types.